src/incremental/__init__.py

```diff
@@ -392,18 +391,34 @@ def _get_setuptools_version(dist):  # type: (_Distribution) -> None
     Setuptools integration: load the version from the working directory
 
     This function is registered as a setuptools.finalize_distribution_options
-    entry point [1]. It is a no-op unless there is a pyproject.toml containing
-    an empty [tool.incremental] section.
-
-    @param dist: An empty C{setuptools.Distribution} instance to mutate.
+    entry point [1]. Consequently, it is called in all sorts of weird
+    contexts. In setuptools, silent failure is the law.
 
     [1]: https://setuptools.pypa.io/en/latest/userguide/extension.html#customizing-distribution-options
+
+    @param dist:
+        A (possibly) empty C{setuptools.Distribution} instance to mutate.
+        There may be some metadata here if a `setup.py` called `setup()`,
+        but this hook is always called before setuptools loads anything
+        from ``pyproject.toml``.
     """
-    config = _load_pyproject_toml("./pyproject.toml")
-    if not config or not config.has_tool_incremental:
+    try:
+        # When operating in a packaging context (i.e. building an sdist or
+        # wheel) pyproject.toml will always be found in the current working
+        # directory.
+        config = _load_pyproject_toml("./pyproject.toml")
+    except Exception:
         return
 
-    dist.metadata.version = _existing_version(config.path).public()
+    if not config.opt_in:
+        return
+
+    try:
+        version = _existing_version(config.version_path)
+    except Exception:
+        return
```

Comment:
This part may be excessive paranoia but I really don't want to inflict any more pain on the ecosystem. It'll sure make debugging difficult for users of Incremental, though.

Reply:
I am not sure what to say about this... but not a blocker.

Maybe is better to break in order to detect and be aware of the exact use cases.

Reply:
Okay, you've talked me down. :) I'll make a missing ``_version.py`` a no-op, but propagate an exception that results from a syntax error.